wordpress-mobile/WordPress-Android
# QRCode Auth: Add tracking events

Parent #16481 

This PR adds tracking events to the QRCodeAuthFlow. 
- Matches the iOS work here https://github.com/wordpress-mobile/WordPress-iOS/pull/18766
- Has me considering refactoring all the class names so they match the events, would love opinions.
- Needs a discussion with the iOS team to decide if we should cover error events and such, and if so, they 
will be added in a future PR.

**Merge Instructions**
- Ensure that #16713  has been merged to the feature branch
- Merge as normal

**Prior to testing this PR you will need to**
- Make sure your **real device** is sandboxed 
- Make sure your sandbox is pointed to D79766-code (apply the patch)
- Have access to the "Calypso live (direct link)" in the related Calypso PR: 
    - https://github.com/Automattic/wp-calypso/pull/63257

### To test:
**Pre-req**
- Install the app
- Login using a WP.com account
- Navigate to Me -> App Settings -> Debug Settings
- Enable `QRCodeAuthFlowFeatureConfig`
- Restart the App

#### Testing from the menu
1. Launch the app
2. Navigate to My Site > Me > Scan Login Code
3. ✅ Verify you see: `🔵 Tracked: qrlogin_scanner_displayed, Properties: {"origin":"menu"}`
4. Tap the X to dismiss the flow
5. ✅  Verify you see: `🔵 Tracked: qrlogin_scanned_dismissed, Properties: {"origin":"menu"}`
6. Scan a valid login code
7. ✅  Verify you see: `🔵 Tracked: qrlogin_scanner_scanned_code, Properties: {"origin":"menu"}`
8. ✅ Verify you see `🔵 Tracked: qrlogin_verify_displayed <origin: menu>`
9. Once it's done loading, ✅ `🔵 Tracked: qrlogin_verify_token_validated, Properties: {"origin":"menu"}`
10. Tap cancel, ✅  `🔵 Tracked: qrlogin_verify_cancelled, Properties: {"origin":"menu"}`
11. Scan again, and tap 'Yes, log me in'
12.✅  Verify you see `🔵 Tracked: qrlogin_verify_approved, Properties: {"origin":"menu"}`
13. Once it's done loading, .✅ `🔵 Tracked: qrlogin_authenticated, Properties: {"origin":"menu"}`
14. Tap dismiss,.✅  `🔵 Tracked: qrlogin_verify_dismissed, Properties: {"origin":"menu"}`


## Regression Notes
1. Potential unintended areas of impact N/A
2. What I did to test those areas of impact (or what existing automated tests I relied on) N/A
3. What automated tests I added (or what prevented me from doing so)
QRCodeAuthViewModelTest

PR submission checklist:
